packages/core/src/browser/style/tabs.css

```diff
@@ -54,10 +54,12 @@
 }
 
```

Comment:
With the CSS adaptions as they are here there is one downside: The place for the X is reserved, even when where are not hovering.

I played around and replaced all of the CSS modifications done here with these simple rules:

```
.p-TabBar-tab .theia-tab-icon-label {
  flex: 1;
}

.p-TabBar-tab.p-mod-closable:hover .theia-tab-icon-label {
  overflow: hidden;
}
```
The first rule lets the label grow as much as possible, pushing the X to the right.
The second rule cuts of the label on hover, making room for the X.

What do you think? Did I miss something?